openedx/features/job_board/tests/views/test_job_board.py

```diff
@@ -0,0 +1,41 @@
+from django.contrib.sites.models import Site
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+
+from openedx.core.djangoapps.theming.models import SiteTheme
+from openedx.features.job_board.views import JobCreateView, JobListView
+from openedx.features.job_board.tests.factories import JobFactory
+
+
+class JobBoardViewTest(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(JobBoardViewTest, cls).setUpClass()
+        site = Site(domain='testserver', name='test')
+        site.save()
+        theme = SiteTheme(site=site, theme_dir_name='philu')
+        theme.save()
+
+    def test_job_create_view(self):
+        response = self.client.get(reverse('job_create'))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(JobCreateView.fields, '__all__')
+
+    def test_job_detail_view_invalid_pk(self):
+        response = self.client.get(reverse('job_detail', kwargs={'pk': 1}))
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+
+    def test_job_detail_view_valid_pk(self):
+        job = JobFactory()
+        job.save()
+        response = self.client.get(reverse('job_detail', kwargs={'pk': 1}))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_job_list_view(self):
+        response = self.client.get(reverse('job_list'))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(JobListView.paginate_by, 10)
+        self.assertEqual(JobListView.ordering, ['-created'])
+
```

Comment:
As per my understanding, we are using `CreateView` for job create. So it trigger two API calls `GET` and `POST`. You have written one test of `GET` (`test_job_create_view`). One more test is required for `POST` which will test data is submitted to `reverse('job_create')`

More test required:
1. Test for successful form submission `POST` call
2. Test for unsuccessful form submission `POST` call
3. Test to assert that job urls are accessible without login/authentication

Reply:
Added tests for these scenarios